scripts/compile_coq.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/sh -e
+
+help() {
+  echo "./compile-coq <compiler path> <path to directory> [--opam opam] [-c|--compile-opam] [-r|--remove]"
+  echo "[compiler path]     is the path to the ocaml compiler"
+  echo "[path to directory] is the path to the drectory where eveything will be build." 
+  echo "                    This script considers that it is free to do anything from"
+  echo "                    this directory, including deleting it."
+  echo "                    Omitting this entry will result in using a temp dir."
+  echo "[-c|--compile-opam] Compile and use the latest version of opam."
+  echo "[--opam opam]       Use a specific version of opam."
+  echo "[-r|--remove]       Remove the destination directory after coq is compiled."
+}
+
+#
+# Argument parsing
+#
+
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+  -c|--compile-opam)
+    COMPILE_OPAM=true
+    shift # past argument
+    ;;
+  --opam)
+    COMPILE_OPAM=false
+    OPAM="$2"
+    shift # past argument
+    shift # past value
+    ;;
+  -r|--remove)
+    REMOVE=true
+    shift # past argument
+    ;;
+  *)    # unknown option
+    POSITIONAL+=("$1") # save it in an array for later
+    shift # past argument
+    ;;
+esac
+done
+set -- "${POSITIONAL[@]}" # restore positional parameters
+
+if [ "$#" -le 0 ]
+then
+  help
+  exit 0
+elif [ "$#" -le 1 ]
+then
+  COMPILER_TO_TEST=$(realpath $1)
+  CWD=$(mktemp -d)
+else
+  COMPILER_TO_TEST=$(realpath $1)
+  CWD=$(realpath $2)
+fi
+
+# Setup the working environment
+mkdir -p $CWD
+OCAML_PATH=$CWD/ocaml
+export OPAMROOT=$CWD/.opam
+rm -rdf $OPAMROOT
+
+# If requested remove the working dir on the exit
+remove_dir_trap() {
+  if [ "$REMOVE" = true ] ; then
+    rm -rfd $CWD
+  fi
+}
+
+trap "remove_dir_trap" EXIT
+
+# Compile opam if needed
+if [ "$COMPILE_OPAM" = true ] ; then
+  OPAM_PATH=$CWD/opam
+  OPAM_INSTALL=$OPAM_PATH/_install
+  OPAM=$OPAM_INSTALL/bin/opam
+
+  git clone https://github.com/ocaml/opam $OPAM_PATH
+  pushd $OPAM_PATH
+
+  ./configure --prefix=$OPAM_INSTALL
+  make lib-ext
+  make
+  make install
+  popd
+elif [ -z "${OPAM}" ] ; then
+  OPAM=opam
+else
+  true
+fi
+
+echo "Using opam $OPAM"
+
+# Copy the compiler to the working directory
+rm -rdf $OCAML_PATH
+cp -r $COMPILER_TO_TEST $OCAML_PATH
+
+$OPAM init -n --disable-sandboxing --bare
```

Comment:
Won't that fail if opam has already been init-ed ? Or at the very least mess up the user's configuration of opam (e.g. disable opam's sandboxing for all uses of opam after execution of this script)

Reply:
No! I'm using a trick where I'm exporting a OPAMROOT set to a new folder and according to `opam`:
```       
--root=ROOT
           Use ROOT as the current root path. This is equivalent to setting
           $OPAMROOT to ROOT.
```
So all of opam's invocation are done from a different opam directory than the normal one.

Reply:
Ok, great ! might be worth a small comment just in case.